### PR TITLE
WFCORE-993 fix embedded HC crash 

### PR DESCRIPTION
**Pull request by luck3y**
Caused by reload dropping --admin-only

### CODE REVIEW

File: cli/src/main/java/org/jboss/as/cli/handlers/ReloadHandler.java [hunk 1]
```diff
@@ -167,7 +167,17 @@ protected void doHandle(CommandContext ctx) throws CommandLineException {
 
     private void doHandleEmbedded(CommandContext ctx, ModelControllerClient client) throws CommandLineException {
 
+        assert(embeddedServerRef != null);
+        assert(embeddedServerRef.get() != null);
+
         final ModelNode op = this.buildRequestWithoutHeaders(ctx);
+        if (embeddedServerRef.get().isHostController()) {
```
Comment (bstansberry) [inline cli/src/main/java/org/jboss/as/cli/handlers/ReloadHandler.java]: WDYT about adding some validation in this block?

```
final ParsedCommandLine args = ctx.getParsedCommandLine();
if (this.adminOnly.isPresent(args) && "FALSE".equalsIgnoreCase(this.adminOnly.getValue(args)) {
 throw someExceptionCauseWeCantDoWhatTheyRequested;
}
```

Bonus is overriding the canAppearNext() method of adminOnly so it returns false if embeddedServerRef.get() != null && embeddedServerRef.get().isHostController(). That will remove that param from tab completion.

Comment (luck3y) [inline cli/src/main/java/org/jboss/as/cli/handlers/ReloadHandler.java]: Sure, good idea I'll add.

Comment (bstansberry) [inline cli/src/main/java/org/jboss/as/cli/handlers/ReloadHandler.java]: The docs on the --admin-only param in reload.txt should explain all this too; param won't be offered, value will always be true.

**Comment by luck3y** [comment]
Note this is also disabling /host=master:reload() due to that being able to force admin-only=false also. I can probably alter that if we really want /host=master:reload() to work in embedded, but it seems to be problematic (especially in .cli scripts, as it returns immediatly before the HC is actually restarted).

**Comment by luck3y** [comment]
Forgot docs, updating now.

**Comment by luck3y** [comment]
Hrm, thats not quite what I wanted ;) Fixing.

**Pull request merged**